pkg/controller/machine-set-boot-image/platform_helpers.go

```diff
@@ -0,0 +1,183 @@
+func checkMachineSet(infra *osconfigv1.Infrastructure, machineSet *machinev1beta1.MachineSet, configMap *corev1.ConfigMap, arch string) (bool, *machinev1beta1.MachineSet, error) {
+	switch infra.Status.PlatformStatus.Type {
+	case osconfigv1.AWSPlatformType:
+		return reconcileAWS(machineSet, configMap, arch)
+	case osconfigv1.AzurePlatformType:
+		return reconcileAzure(machineSet, configMap, arch)
+	case osconfigv1.BareMetalPlatformType:
+		return reconcileBareMetal(machineSet, configMap, arch)
+	case osconfigv1.AlibabaCloudPlatformType:
+		return reconcileAliBaba(machineSet, configMap, arch)
+	case osconfigv1.OpenStackPlatformType:
+		return reconcileOpenStack(machineSet, configMap, arch)
+	case osconfigv1.EquinixMetalPlatformType:
+		return reconcileEquinixMetal(machineSet, configMap, arch)
+	case osconfigv1.GCPPlatformType:
+		return reconcileGCP(machineSet, configMap, arch)
+	case osconfigv1.KubevirtPlatformType:
+		return reconcileKubevirt(machineSet, configMap, arch)
+	case osconfigv1.IBMCloudPlatformType:
+		return reconcileIBMCCloud(machineSet, configMap, arch)
+	case osconfigv1.LibvirtPlatformType:
+		return reconcileLibvirt(machineSet, configMap, arch)
+	case osconfigv1.VSpherePlatformType:
+		return reconcileVSphere(machineSet, configMap, arch)
+	case osconfigv1.NutanixPlatformType:
+		return reconcileNutanix(machineSet, configMap, arch)
+	case osconfigv1.OvirtPlatformType:
+		return reconcileOvirt(machineSet, configMap, arch)
+	case osconfigv1.ExternalPlatformType:
+		return reconcileExternal(machineSet, configMap, arch)
+	case osconfigv1.PowerVSPlatformType:
+		return reconcilePowerVS(machineSet, configMap, arch)
+	case osconfigv1.NonePlatformType:
+		return reconcileNone(machineSet, configMap, arch)
+	default:
+		return unmarshalToFindPlatform(machineSet, configMap, arch)
+	}
+}
+
+func reconcileAWS(machineSet *machinev1beta1.MachineSet, _ *corev1.ConfigMap, arch string) (patchRequired bool, newMachineSet *machinev1beta1.MachineSet, err error) {
+	klog.Infof("Skipping machineset %s, unsupported platform type AWS with %s arch", machineSet.Name, arch)
+	return false, nil, nil
+}
+
+func reconcileAzure(machineSet *machinev1beta1.MachineSet, _ *corev1.ConfigMap, arch string) (patchRequired bool, newMachineSet *machinev1beta1.MachineSet, err error) {
+	klog.Infof("Skipping machineset %s, unsupported platform type Azure with %s arch", machineSet.Name, arch)
+	return false, nil, nil
+}
+
+func reconcileBareMetal(machineSet *machinev1beta1.MachineSet, _ *corev1.ConfigMap, arch string) (patchRequired bool, newMachineSet *machinev1beta1.MachineSet, err error) {
+	klog.Infof("Skipping machineset %s, unsupported platform type BareMetal with %s arch", machineSet.Name, arch)
+	return false, nil, nil
+}
+
+func reconcileAliBaba(machineSet *machinev1beta1.MachineSet, _ *corev1.ConfigMap, arch string) (patchRequired bool, newMachineSet *machinev1beta1.MachineSet, err error) {
```

Comment:
I think we're looking to remove this as a platform: https://github.com/openshift/machine-config-operator/pull/4112

Not sure if we should just leave it out here as well

Reply:
will remove thanks!